proof_of_stake/src/tests.rs

```diff
@@ -1930,30 +1948,55 @@ fn advance_epoch(s: &mut TestWlStorage, params: &PosParams) -> Epoch {
 
 fn arb_genesis_validators(
     size: Range<usize>,
+    threshold: Option<token::Amount>,
 ) -> impl Strategy<Value = Vec<GenesisValidator>> {
     let tokens: Vec<_> = (0..size.end)
         .map(|_| (1..=10_000_000_u64).prop_map(token::Amount::from))
         .collect();
-    (size, tokens).prop_map(|(size, token_amounts)| {
-        // use unique seeds to generate validators' address and consensus key
-        let seeds = (0_u64..).take(size);
-        seeds
-            .zip(token_amounts)
-            .map(|(seed, tokens)| {
-                let address = address_from_simple_seed(seed);
-                let consensus_sk = common_sk_from_simple_seed(seed);
-                let consensus_key = consensus_sk.to_public();
-
-                let commission_rate = Decimal::new(5, 2);
-                let max_commission_rate_change = Decimal::new(1, 2);
-                GenesisValidator {
-                    address,
-                    tokens,
-                    consensus_key,
-                    commission_rate,
-                    max_commission_rate_change,
+    let tokens = tokens.prop_map(move |mut amounts| {
+        if let Some(thresh) = threshold {
+            if !amounts.iter().any(|&x| x >= thresh) {
+                let mut rng = rand::thread_rng();
```

Comment:
Having the rng here that's not seeded from proptest makes this non-deterministic and that could be a hindrance for possible issues caught by the tests that are using this. I think we can write this slightly differently - I'll push a fixup

Reply:
[3100bfc](https://github.com/anoma/namada/pull/1576/commits/3100bfc41155c19aaf7f996245acba4de92c4281)

Reply:
ah good point, thx, didn't consider this before